src/logging.rs

```diff
@@ -1,48 +1,76 @@
-use slog::*;
+use redislog::Builder;
+use slog::{error, o, Drain, Duplicate, Level, Logger, OwnedKVList, Record};
+use slog_async::Async;
 use slog_redis as redislog;
-use std::result::Result;
+use slog_term::{CompactFormat, TermDecorator};
+use std::{fmt::Display, result::Result};
+
+#[derive(Clone)]
+struct FallbackToStderr<D: Drain> {
+    drain: D,
+}
+
+impl<D: Drain> Drain for FallbackToStderr<D>
+where
+    D::Err: Display,
+{
+    type Ok = ();
+    type Err = ();
+    fn log(&self, record: &Record, logger_values: &OwnedKVList) -> Result<(), ()> {
+        if let Err(err) = self.drain.log(record, logger_values) {
+            eprint!("A drain cloud not log to its destenation: {}", err);
+        }
+        Ok(())
+    }
+
+    #[inline]
+    fn is_enabled(&self, level: Level) -> bool {
+        self.drain.is_enabled(level)
+    }
+}
 
 pub fn create_logger(arg_matches: &ArgMatches) -> Result<slog::Logger, String> {
-    let min_log_level = match arg_matches.occurrences_of(args::VERBOSITY) {
-        0 => slog::Level::Warning,
-        1 => slog::Level::Info,
-        2 => slog::Level::Debug,
-        _ => slog::Level::Trace,
+    let min_log_level = match arg_matches.occurrences_of(VERBOSITY) {
+        0 => Level::Warning,
+        1 => Level::Info,
+        2 => Level::Debug,
+        _ => Level::Trace,
     };
 
-    let min_log_level = match arg_matches.value_of(args::LOG_LEVEL) {
+    let min_log_level = match arg_matches.value_of(LOG_LEVEL) {
         Some(level) => match level.parse::<args::LogLevelType>()? {
-            args::LogLevelType::error => slog::Level::Error,
-            args::LogLevelType::warn => slog::Level::Warning,
-            args::LogLevelType::info => slog::Level::Info,
-            args::LogLevelType::debug => slog::Level::Debug,
-            args::LogLevelType::trace => slog::Level::Trace,
+            args::LogLevelType::error => Level::Error,
+            args::LogLevelType::warn => Level::Warning,
+            args::LogLevelType::info => Level::Info,
+            args::LogLevelType::debug => Level::Debug,
+            args::LogLevelType::trace => Level::Trace,
         },
         None => min_log_level,
     };
 
-    let decorator = slog_term::TermDecorator::new().force_color().build();
-    let term_drain = slog_term::FullFormat::new(decorator)
+    let decorator = TermDecorator::new().force_color().build();
+    let term_drain = CompactFormat::new(decorator)
         .build()
         .filter_level(min_log_level)
+        .map(|drain| FallbackToStderr { drain })
         .fuse();
 
     let mut err: Option<String> = None;
     let drain = match redis_logger(&arg_matches) {
         Ok(Some(redis_logger)) => {
-            let both = slog::Duplicate::new(redis_logger, term_drain).fuse();
-            slog_async::Async::new(both.filter_level(min_log_level).fuse())
-                .build()
-                .fuse()
+            let both = Duplicate::new(redis_logger, term_drain).fuse();
```

Comment:
We don't have to `fuse` or `ignore_res` for the non-terminal ones here I think i.e. just remove `fuse`.

Reply:
We need to use `FallbackToStderr` for the redislogger though...

Reply:
We use it, it is already returned like that from the `redis_logger(&arg_matches)`

Reply:
Ah, OK missed that one.